agent/client/agent_http_client.go

```diff
@@ -142,11 +142,19 @@ func CreateStandardAgentHttpRequest(method, requestUrl string, body io.Reader) (
 }
 
 func RetryErrorHandler(resp *http.Response, err error, attempt int) (retryResponse *http.Response, retryError error) {
-	if resp != nil && resp.StatusCode < 600 && resp.StatusCode >= 100 {
```

Comment:
I think it is still worth it keeping the validation for statusCode?

Reply:
Given that is the only possible range of status codes, I believe it is redundant to check and add validations for it